src/gun_http2.erl

```diff
@@ -290,41 +294,42 @@ frame(State=#http2_state{http2_machine=HTTP2Machine0}, Frame, CookieStore, EvHan
 			{maybe_ack_or_notify(State#http2_state{http2_machine=HTTP2Machine}, Frame),
 				CookieStore, EvHandlerState};
 		{ok, {data, StreamID, IsFin, Data}, HTTP2Machine} ->
-			{StateRet, CookieStoreRet, EvHandlerStateRet} = data_frame(
-				State#http2_state{http2_machine=HTTP2Machine}, StreamID, IsFin, Data,
-				CookieStore, EvHandler, EvHandlerState),
-			case StateRet of
-				{state, State1} -> {State1, CookieStoreRet, EvHandlerStateRet};
-				Error -> {Error, CookieStoreRet, EvHandlerStateRet}
-			end;
+			data_frame(State#http2_state{http2_machine=HTTP2Machine},
+				StreamID, IsFin, Data, CookieStore, EvHandler, EvHandlerState);
 		{ok, {headers, StreamID, IsFin, Headers, PseudoHeaders, BodyLen}, HTTP2Machine} ->
-			headers_frame(State#http2_state{http2_machine=HTTP2Machine},
+			{StateRet, CookieStoreRet, EvHandlerStateRet} = headers_frame(
+				State#http2_state{http2_machine=HTTP2Machine},
```

Comment:
This is a general comment regarding both data_frame, headers_frame and the other *_frame functions: they should follow the same return value format. So if data_frame changes to a tuple return value, they all should.

Reply:
Updating the following functions return state tuple or error: `headers_frame`, `trailers_frame`, `rst_stream_frame`, `headers_frame_inform`, `headers_frame_connect`, `headers_frame_connect_websocket`, `headers_frame_response`, `ignored_frame`.